cmd/snap-update-ns/update.go

```diff
@@ -93,8 +93,14 @@ func executeMountProfileUpdate(upCtx MountProfileUpdateContext) error {
 	// Compute the new current profile so that it contains only changes that were made
 	// and save it back for next runs.
 	var currentAfter osutil.MountProfile
+	for i := len(changesMade) - 1; i >= 0; i-- {
```

Comment:
I think it's useful to leave a comment for ourselves that the initial state was initially iterated over from back to front, so the things we keep are also present in reverse order. Thus order to preserve the stable state for kept entries we should record them in reverse order.

Reply:
Pushed now